pkgs/development/python-modules/debugpy/default.nix

```diff
@@ -0,0 +1,68 @@
+{ stdenv, buildPythonPackage, fetchFromGitHub
+, substituteAll, gdb
+, colorama, django, flask, gevent, psutil, pytest
+, pytest-timeout, pytest_xdist, requests
+, isPy27
+}:
+
+buildPythonPackage rec {
+  pname = "debugpy";
+  version = "1.0.0b11";
+
+  src = fetchFromGitHub {
+    owner = "Microsoft";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1zhar63wl05i15sgdkqmx4j7h6amjmmciz8q4ncyapkzipy9z6bq";
+  };
+
+  patches = [
+    # Hard code GDB path (used to attach to process)
+    (substituteAll {
+      src = ./hardcode-gdb.patch;
+      inherit gdb;
+    })
+
+    # Fix importing debugpy in:
+    # - test_nodebug[module-launch(externalTerminal)]
+    # - test_nodebug[module-launch(integratedTerminal)]
+    #
+    # NOTE: The import failures seen in these tests without the patch
+    # will be seen if a user "installs" debugpy by adding it to PYTHONPATH.
+    # To avoid this issue, debugpy should be installed using python.withPackages:
+    # python.withPackages (ps: with ps; [ debugpy ])
+    ./fix-test-pythonpath.patch
+  ];
+
+  # Remove pre-compiled "attach" libraries and recompile for host platform
+  # Compile flags taken from linux_and_mac/compile_linux.sh & linux_and_mac/compile_mac.sh
+  preBuild = ''(
+    set -x
+    cd src/debugpy/_vendored/pydevd/pydevd_attach_to_process
+    rm *.so *.dylib *.dll *.exe *.pdb
+    ${stdenv.cc}/bin/c++ linux_and_mac/attach.cpp -Ilinux_and_mac -fPIC -nostartfiles ${{
+      "x86_64-linux"  = "-shared -m64 -o attach_linux_amd64.so";
+      "i686-linux"    = "-shared -m32 -o attach_linux_x86.so";
+      "x86_64-darwin" = "-std=c++11 -lc -D_REENTRANT -dynamiclib -arch x86_64 -o attach_x86_64.dylib";
+      "i686-darwin"   = "-std=c++11 -lc -D_REENTRANT -dynamiclib -arch i386 -o attach_x86.dylib";
+    }.${stdenv.hostPlatform.system}}
+  )'';
+
+  checkInputs = [
+    colorama django flask gevent psutil pytest
+    pytest-timeout pytest_xdist requests
+  ];
+
+  # Override default arguments in pytest.ini
+  checkPhase = "pytest --timeout 0 -n $NIX_BUILD_CORES"
+               # gevent fails to import zope.interface with Python 2.7
+               + stdenv.lib.optionalString isPy27 " -k 'not test_gevent'";
```

Comment:
Does anyone have any insight on why this could be failing for Python 2.7, but not Python 3.8? Based on the 
[test log](https://github.com/NixOS/nixpkgs/files/4832682/tests-260.log) zope.interface is included in `PYTHONPATH`. Could it be related to zope.interface using nested directories without `__init__.py`?

Reply:
zope uses native namespaces. Which don't work in python2. https://www.python.org/dev/peps/pep-0420/ if you're more interested

Reply:
Hmm, apparently zope should support Python 2.7: [https://pypi.org/project/zope.interface/](https://pypi.org/project/zope.interface). Maybe it hasn't actually been tested in 2.7? gevent is an [optional feature](https://github.com/microsoft/debugpy/wiki/Debug-configuration-settings), so I'm not too worried about it being unsupported in Python 2.7.

Reply:
zope works fine in python 2.7, but you need to do something similar to https://nixos.org/nixpkgs/manual/#python.buildenv-function for it to work properly in nix.

I would say what you did is fine

Reply:
also, python2 has been EOL for 1/2 a year now, with active CVES. Personally I think it's a waste of time to try and support it, especially for new packages